src/Markdig/Extensions/Bootstrap/BootstrapExtension.cs

```diff
@@ -61,7 +61,10 @@ private static void PipelineOnDocumentProcessed(MarkdownDocument document)
                     var attributes = node.GetAttributes();
                     attributes.AddClass("alert");
                     attributes.AddProperty("role", "alert");
-                    string? @class = alertBlock.Kind.AsSpan() switch
+
+                    Span<char> upperKind = new char[alertBlock.Kind.Length];
```

Comment:
Similarly, we can use stackalloc here and protect it under a check `kind.Length >= 16`

Reply:
I didn't use `stackallock` here for a reason. It is only freed after the method returns so it might lead to a temporary memory leak in case of huge documents with many alert blocks.

Reply:
You could switch on `alertBlock.Kind.ToUpperInvariant()` instead (avoids the allocation if it's already upper case).
Or move the `stackalloc char[16]` out of the loop